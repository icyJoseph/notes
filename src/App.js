import React, { Component } from "react";

class App extends Component {
  render() {
    return <div>My Notes</div>;
  }
}

export default App;
