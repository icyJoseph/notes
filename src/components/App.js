import React, { Component } from "react";
import { Form, Button, FormControl } from "react-bootstrap";

class App extends Component {
  render() {
    return <div>My Notes</div>;
  }
}

export default App;
