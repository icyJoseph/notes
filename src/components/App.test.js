import React from "react";
import { shallow } from "enzyme";

import App from "./App";

describe("App", () => {
  const app = shallow(<App />);
  it("renders correctly", () => {
    expect(app).toMatchSnapshot();
  });

  it("renders `My Notes` title", () => {
    expect(app.find(".title").exists()).toBe(true);
    expect(app.find(".title").text()).toEqual("My Notes");
  });

  it("renders a bootstrap Form", () => {
    expect(app.find("Form").exists()).toBe(true);
  });

  it("renders a bootstrap FormControl", () => {
    expect(app.find("FormControl").exists()).toBe(true);
  });

  it("renders a bootstrap Button with `Submit` text", () => {
    expect(app.find("Button").exists()).toBe(true);
    expect(
      app
        .find("Button")
        .render()
        .text()
    ).toEqual("Submit");
  });

  describe("App state", () => {
    let initialAppState = { text: "" };
    it("has a state object", () => {
      expect(app.state()).toEqual(initialAppState);
    });

    describe("input is controlled", () => {
      let note = "Testing FormControl input";
      beforeEach(() => {
        app.find("FormControl").simulate("change", { target: { value: note } });
      });

      it("updates state with input", () => {
        expect(app.state().text).toEqual(note);
      });

      it("shows `state.text` as its value", () => {
        expect(app.find("FormControl").get(0).props.value).toEqual(
          app.state().text
        );
      });
    });
  });
});
