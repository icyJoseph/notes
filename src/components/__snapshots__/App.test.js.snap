// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="title"
>
          My Notes
</h2>,
        <Form
          bsClass="form"
          componentClass="form"
          horizontal={false}
          inline={true}
          onSubmit={[Function]}
>
          <FormControl
                    bsClass="form-control"
                    componentClass="input"
                    onChange={[Function]}
                    value=""
          />
          <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="default"
                    className="submit-button"
                    disabled={false}
                    onClick={[Function]}
          >
                    Submit
          </Button>
</Form>,
        <ul
          className="notes-list"
>
          
</ul>,
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          className="clear-notes"
          disabled={false}
          onClick={[Function]}
>
          Clear Notes
</Button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "My Notes",
          "className": "title",
        },
        "ref": null,
        "rendered": "My Notes",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "form",
          "children": Array [
            <FormControl
              bsClass="form-control"
              componentClass="input"
              onChange={[Function]}
              value=""
/>,
            <Button
              active={false}
              block={false}
              bsClass="btn"
              bsStyle="default"
              className="submit-button"
              disabled={false}
              onClick={[Function]}
>
              Submit
</Button>,
          ],
          "componentClass": "form",
          "horizontal": false,
          "inline": true,
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "form-control",
              "componentClass": "input",
              "onChange": [Function],
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "active": false,
              "block": false,
              "bsClass": "btn",
              "bsStyle": "default",
              "children": "Submit",
              "className": "submit-button",
              "disabled": false,
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Submit",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "notes-list",
        },
        "ref": null,
        "rendered": Array [],
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "block": false,
          "bsClass": "btn",
          "bsStyle": "default",
          "children": "Clear Notes",
          "className": "clear-notes",
          "disabled": false,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Clear Notes",
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="title"
>
            My Notes
</h2>,
          <Form
            bsClass="form"
            componentClass="form"
            horizontal={false}
            inline={true}
            onSubmit={[Function]}
>
            <FormControl
                        bsClass="form-control"
                        componentClass="input"
                        onChange={[Function]}
                        value=""
            />
            <Button
                        active={false}
                        block={false}
                        bsClass="btn"
                        bsStyle="default"
                        className="submit-button"
                        disabled={false}
                        onClick={[Function]}
            >
                        Submit
            </Button>
</Form>,
          <ul
            className="notes-list"
>
            
</ul>,
          <Button
            active={false}
            block={false}
            bsClass="btn"
            bsStyle="default"
            className="clear-notes"
            disabled={false}
            onClick={[Function]}
>
            Clear Notes
</Button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "My Notes",
            "className": "title",
          },
          "ref": null,
          "rendered": "My Notes",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form",
            "children": Array [
              <FormControl
                bsClass="form-control"
                componentClass="input"
                onChange={[Function]}
                value=""
/>,
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="default"
                className="submit-button"
                disabled={false}
                onClick={[Function]}
>
                Submit
</Button>,
            ],
            "componentClass": "form",
            "horizontal": false,
            "inline": true,
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "form-control",
                "componentClass": "input",
                "onChange": [Function],
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "active": false,
                "block": false,
                "bsClass": "btn",
                "bsStyle": "default",
                "children": "Submit",
                "className": "submit-button",
                "disabled": false,
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Submit",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "notes-list",
          },
          "ref": null,
          "rendered": Array [],
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "block": false,
            "bsClass": "btn",
            "bsStyle": "default",
            "children": "Clear Notes",
            "className": "clear-notes",
            "disabled": false,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Clear Notes",
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
